schema/schema.go

```diff
@@ -0,0 +1,50 @@
+package schema
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var (
+	// fs stores the embedded http.FileSystem
+	// having the OCI JSON schema files in root "/".
+	fs = _escFS(false)
+
+	specs = map[jsonManifest]string{
+		MediaTypeManifest:     "image-manifest-schema.json",
+		MediaTypeManifestList: "manifest-list-schema.json",
+	}
+)
+
+// FindValidator returns the validator for the given media type
+// or an error if none could be found.
+func FindValidator(mediaType string) (Validator, error) {
```

Comment:
Indeed, many thanks for the pointer! I'll reiterate on this, my approach is a bit different than `crypto.Hash` but looking over it I like it much better.